File: content/code/r/collection.r
args <- commandArgs(trailingOnly = TRUE)
folder_path <- args[1]

data_array <- list()
files <- list.files(folder_path)

for (filename in files) {
if (grepl("^result_[0-9]+\\.rds$", filename)) {
    index <- as.numeric(sub("^result_([0-9]+)\\.rds$", "\\1", filename))
    file_path <- file.path(folder_path, filename)
    data_array[[index]] <- readRDS(file_path)
}
}

# Write the data array to a result file
result_file_path <- file.path(folder_path, "result.rds")
saveRDS(data_array, result_file_path)

File: content/code/python/collection.py
import os
import pickle
import sys

folder_path = sys.argv[1]

data_array = []
for filename in os.listdir(folder_path):
    if filename.startswith("result_"):
        index = int(filename.split("_")[1].split(".")[0])
        file_path = os.path.join(folder_path, filename)
        with open(file_path, "rb") as f:
            data_array.insert(index, pickle.load(f))

with open(file_path, "wb") as f:
    pickle.dump(data_array, f)
